Assets/MixedRealityToolkit/_Core/Utilities/Lines/Renderers/ParticleSystemLineRenderer.cs

```diff
@@ -0,0 +1,286 @@
+        {
+            get { return lifetimeAfterDisabled; }
+            set
+            {
+                if (value < 0.01f)
+                {
+                    lifetimeAfterDisabled = 0.01f;
+                }
+                else if (value > 0.5f)
+                {
+                    lifetimeAfterDisabled = 0.5f;
+                }
+                else
+                {
+                    lifetimeAfterDisabled = value;
+                }
+            }
+        }
+
+        [SerializeField]
+        private Gradient decayGradient = new Gradient();
+
+        public Gradient DecayGradient
+        {
+            get { return decayGradient; }
+            set { decayGradient = value; }
+        }
+
+        [SerializeField]
+        [HideInInspector]
+        private ParticleSystem particles;
+
+        [SerializeField]
+        [HideInInspector]
+        private ParticleSystemRenderer mainParticleRenderer;
+
+        public ParticleSystemRenderer MainParticleRenderer
+        {
+            get
+            {
+                if (particles == null)
+                {
+                    particles = gameObject.EnsureComponent<ParticleSystem>();
+                }
+
+                if (mainParticleRenderer == null)
+                {
+                    mainParticleRenderer = particles.EnsureComponent<ParticleSystemRenderer>();
+                }
+
+                return mainParticleRenderer;
+            }
+            set { mainParticleRenderer = value; }
+        }
+
+        private readonly ParticleSystem.Particle[] mainParticleArray = new ParticleSystem.Particle[GlobalMaxParticles];
+
+        private ParticleSystem.NoiseModule mainNoiseModule;
+
+        private float decayStartTime = 0f;
+
+        private void OnEnable()
+        {
+            if (particles == null)
+            {
+                particles = gameObject.EnsureComponent<ParticleSystem>();
+            }
+
+            mainNoiseModule = particles.noise;
+
+            ParticleSystem.EmissionModule emission = particles.emission;
+            emission.rateOverTime = new ParticleSystem.MinMaxCurve(0);
+            emission.rateOverDistance = new ParticleSystem.MinMaxCurve(0);
+            emission.enabled = true;
+
+            ParticleSystem.MainModule main = particles.main;
+            main.loop = false;
+            main.playOnAwake = false;
+            main.maxParticles = Mathf.Min(maxParticles, GlobalMaxParticles);
+            main.simulationSpace = ParticleSystemSimulationSpace.World;
+
+            ParticleSystem.ShapeModule shape = particles.shape;
+            shape.enabled = false;
+
+            MainParticleRenderer.sharedMaterial = lineMaterial;
+            MainParticleRenderer.enabled = true;
+
+            // Initialize our particles
+            for (int i = 0; i < mainParticleArray.Length; i++)
+            {
+                ParticleSystem.Particle particle = mainParticleArray[i];
+                particle.startColor = Color.white;
```

Comment:
The particles initial values should be configurable.

Reply:
This is just an initializer. I'm pretty sure the color is set below, but I'll double check.

Reply:
The values are updated correctly in the update loop. The initial colors are just to make sure the system is setup correctly. This and all the other line renderers are executed in edit mode, so the colors, and other values are updated as you changes them.